src/Microsoft.NetCore.Analyzers/Core/Security/DoNotDisableCertificateValidation.cs

```diff
@@ -0,0 +1,198 @@
+
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+                DiagnosticId,
+                s_Title,
+                s_Message,
+                DiagnosticCategory.Security,
+                DiagnosticHelpers.DefaultDiagnosticSeverity,
+                isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+                description: s_Description,
+                helpLinkUri: null,
+                customTags: WellKnownDiagnosticTags.Telemetry);
+        
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public sealed override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+
+            // Security analyzer - analyze and report diagnostics on generated code.
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+            
+            context.RegisterCompilationStartAction(
+                (CompilationStartAnalysisContext compilationStartAnalysisContext) =>
+                {
+                    var compilation = compilationStartAnalysisContext.Compilation;
+                    var systemNetSecurityRemoteCertificateValidationCallbackTypeSymbol = compilation.GetTypeByMetadataName(
+                            WellKnownTypes.SystemNetSecurityRemoteCertificateValidationCallback);
+                    var obj = WellKnownTypes.Object(compilation);
+                    var x509Certificate = WellKnownTypes.X509Certificate(compilation);
+                    var x509Chain = WellKnownTypes.X509Chain(compilation);
+                    var sslPolicyErrors = WellKnownTypes.SslPolicyErrors(compilation);
+
+                    if (systemNetSecurityRemoteCertificateValidationCallbackTypeSymbol == null)
+                    {
+                        return;
+                    }
+                    
+                    compilationStartAnalysisContext.RegisterOperationAction(
+                        (OperationAnalysisContext operationAnalysisContext) =>
+                        {
+                            var delegateCreationOperation =
+                                (IDelegateCreationOperation)operationAnalysisContext.Operation;
+
+                            if (systemNetSecurityRemoteCertificateValidationCallbackTypeSymbol.Equals(delegateCreationOperation.Type))
+                            {
+                                var alwaysReturnTrue = false;
+
+                                switch (delegateCreationOperation.Target.Kind)
+                                {
+                                    case OperationKind.AnonymousFunction:
+                                        var delegateTargetFunction = (IAnonymousFunctionOperation)delegateCreationOperation.Target;
+
+                                        if (delegateTargetFunction == null)
```

Comment:
I believe this cannot be null.

Reply:
what if there are some erroneous code or in the middle of coding, "ServicePointManager.ServerCertificateValidationCallback = new RemoteCertificateValidationCallback();"

Reply:
We generate an `IInvalidOperation` for that case. If you get an `IDelegateCreationOperation` in the operation tree, its Target is guaranteed to be non-null.